src/main/java/stirling/software/SPDF/SPdfApplication.java

```diff
@@ -23,21 +24,41 @@
 import stirling.software.SPDF.config.ConfigInitializer;
 import stirling.software.SPDF.model.ApplicationProperties;
 
+
 @SpringBootApplication
 @EnableScheduling
 public class SPdfApplication {
 
     private static final Logger logger = LoggerFactory.getLogger(SPdfApplication.class);
 
     @Autowired private Environment env;
-
     @Autowired ApplicationProperties applicationProperties;
 
     private static String serverPortStatic;
 
     @Value("${server.port:8080}")
     public void setServerPortStatic(String port) {
-        SPdfApplication.serverPortStatic = port;
+        if (port.equalsIgnoreCase("auto")) {
+            SPdfApplication.serverPortStatic = findAvailablePort(8080);  // Start from port 8080
+        } else {
+            SPdfApplication.serverPortStatic = port;
+        }
+    }
+
+    private static String findAvailablePort(int startPort) {
+        int port = startPort;
+        while (!isPortAvailable(port)) {
+            port++;
+        }
+        return String.valueOf(port);
+    }
+
+    private static boolean isPortAvailable(int port) {
+        try (ServerSocket socket = new ServerSocket(port)) {
+            return true;
+        } catch (IOException e) {
+            return false;
+        }
```

Comment:
It's not necessary for the port number to be >=8080, so can we use `ServerSocket(0)` for a random port instead?

Reply:
Thank you for the suggestion! Using ServerSocket(0) for a random port is a good idea when any available port is acceptable. However, I opted to start from port 8080 because it is commonly used in web development. This ensures that, if 8080 is available, it will be chosen first, which aligns with typical developer expectations. Additionally, some systems or configurations might prefer using ports starting from 8080 for consistency in testing and deployment workflows.
That said, I'm happy to adjust if you'd prefer using ServerSocket(0) for more flexibility. Let me know your thoughts!

Reply:
I have updated the functionality. The port starts from 0 instead of 8080.

Reply:
@sbplat

Reply:
> Thank you for the suggestion! Using ServerSocket(0) for a random port is a good idea when any available port is acceptable. However, I opted to start from port 8080 because it is commonly used in web development. This ensures that, if 8080 is available, it will be chosen first, which aligns with typical developer expectations. Additionally, some systems or configurations might prefer using ports starting from 8080 for consistency in testing and deployment workflows. That said, I'm happy to adjust if you'd prefer using ServerSocket(0) for more flexibility. Let me know your thoughts!

I actually didn't think of this perspective and it makes sense. I think it's more intuitive for the default port to start at 8080, and then increment until it finds an available port. Also, I feel like the "auto" option should be called "random".

Reply:
Merged now 😂 but it's a good point I think having auto be 8080 starting and random be 0 is a good solution for both 
Personally I would hate a truly random port and would prefer it rolling from 8080